Bug/Error yii\db\pgsql\Schema::findConstraints()

```
Database Exception – yii\db\Exception
SQLSTATE[42P01]: Undefined table: 7 ERROR: invalid reference to FROM-clause entry for table "ct"
LINE 8: inner join generate_subscripts(ct.conkey, 1) as s on 1=1
^
HINT: There is an entry for table "ct", but it cannot be referenced from this part of the query.
The SQL being executed was: select
a.attname as column_name,
fc.relname as foreign_table_name,
fns.nspname as foreign_table_schema,
fa.attname as foreign_column_name
from
pg_constraint ct
inner join generate_subscripts(ct.conkey, 1) as s on 1=1
inner join pg_class c on c.oid=ct.conrelid
inner join pg_namespace ns on c.relnamespace=ns.oid
inner join pg_attribute a on a.attrelid=ct.conrelid and a.attnum = ct.conkey[s]
left join pg_class fc on fc.oid=ct.confrelid
left join pg_namespace fns on fc.relnamespace=fns.oid
left join pg_attribute fa on fa.attrelid=ct.confrelid and fa.attnum = ct.confkey[s]
where
ct.contype='f'
and c.relname='acc_periode'
and ns.nspname='public'
order by
fns.nspname, fc.relname, a.attnum

Error Info: Array
(
    [0] => 42P01
    [1] => 7
    [2] => ERROR:  invalid reference to FROM-clause entry for table "ct"
LINE 8:     inner join generate_subscripts(ct.conkey, 1) as s on 1=1
                                           ^
HINT:  There is an entry for table "ct", but it cannot be referenced from this part of the query.
)

↵
Caused by: PDOException
SQLSTATE[42P01]: Undefined table: 7 ERROR: invalid reference to FROM-clause entry for table "ct"
LINE 8: inner join generate_subscripts(ct.conkey, 1) as s on 1=1
^
HINT: There is an entry for table "ct", but it cannot be referenced from this part of the query.

in /home/mdmunir/NetBeansProjects/deesoft/development/vendor/yiisoft/yii2/db/Command.php at line 820
```

I am using postgresql.
Error provide when call `$db->getSchema()->getTableSchemas($schemaName)`

postgresql 9.2

```
Database Exception – yii\db\Exception

SQLSTATE[42P01]: Undefined table: 7 ОШИБКА: в элементе предложения FROM неверная ссылка на таблицу "idx"
LINE 5: INNER JOIN generate_subscripts(idx.indkey, 1) AS k ON 1=1
^
HINT: Таблица "idx" присутствует в запросе, но сослаться на неё из этой части запроса нельзя.
The SQL being executed was: SELECT
i.relname as indexname,
pg_get_indexdef(idx.indexrelid, k + 1, TRUE) AS columnname
FROM pg_index idx
INNER JOIN generate_subscripts(idx.indkey, 1) AS k ON 1=1
INNER JOIN pg_class i ON i.oid = idx.indexrelid
INNER JOIN pg_class c ON c.oid = idx.indrelid
INNER JOIN pg_namespace ns ON c.relnamespace = ns.oid
WHERE idx.indisprimary = FALSE AND idx.indisunique = TRUE
AND c.relname = 'lookup' AND ns.nspname = 'public'
ORDER BY i.relname, k
Error Info: Array
(
    [0] => 42P01
    [1] => 7
    [2] => ОШИБКА:  в элементе предложения FROM неверная ссылка на таблицу "idx"
LINE 5: INNER JOIN generate_subscripts(idx.indkey, 1) AS k ON 1=1
                                       ^
HINT:  Таблица "idx" присутствует в запросе, но сослаться на неё из этой части запроса нельзя.
)
↵
Caused by: PDOException

SQLSTATE[42P01]: Undefined table: 7 ОШИБКА: в элементе предложения FROM неверная ссылка на таблицу "idx"
LINE 5: INNER JOIN generate_subscripts(idx.indkey, 1) AS k ON 1=1
^
HINT: Таблица "idx" присутствует в запросе, но сослаться на неё из этой части запроса нельзя.
```

https://github.com/yiisoft/yii2/blob/master/framework/db/pgsql/Schema.php#L262

--- a/framework/db/pgsql/Schema.php
+++ b/framework/db/pgsql/Schema.php
@@ -259,7 +259,10 @@ SQL;
 SELECT
     i.relname as indexname,
     pg_get_indexdef(idx.indexrelid, k + 1, TRUE) AS columnname
-FROM pg_index idx
+FROM (
+  SELECT *, generate_subscripts(indkey, 1) AS k
+  FROM pg_index
+) idx
 INNER JOIN generate_subscripts(idx.indkey, 1) AS k ON 1=1
 INNER JOIN pg_class i ON i.oid = idx.indexrelid
 INNER JOIN pg_class c ON c.oid = idx.indrelid

--- a/framework/db/pgsql/Schema.php
+++ b/framework/db/pgsql/Schema.php
@@ -213,14 +213,15 @@ select
     fns.nspname as foreign_table_schema,
     fa.attname as foreign_column_name
 from
-    pg_constraint ct
-    inner join generate_subscripts(ct.conkey, 1) as s on 1=1
+    (SELECT ct.conrelid, ct.confrelid, ct.conkey, ct.contype, ct.confkey, generate_subscripts(ct.conkey, 1) AS s
+       FROM pg_constraint ct
+    ) AS ct
     inner join pg_class c on c.oid=ct.conrelid
     inner join pg_namespace ns on c.relnamespace=ns.oid
-    inner join pg_attribute a on a.attrelid=ct.conrelid and a.attnum = ct.conkey[s]
+    inner join pg_attribute a on a.attrelid=ct.conrelid and a.attnum = ct.conkey[ct.s]
     left join pg_class fc on fc.oid=ct.confrelid
     left join pg_namespace fns on fc.relnamespace=fns.oid
-    left join pg_attribute fa on fa.attrelid=ct.confrelid and fa.attnum = ct.confkey[s]
+    left join pg_attribute fa on fa.attrelid=ct.confrelid and fa.attnum = ct.confkey[ct.s]
 where
     ct.contype='f'
     and c.relname={$tableName}

